app/update.go

```diff
@@ -67,3 +290,111 @@ func (m *model) updatePaginator() {
 	}
 	m.list.Styles.PaginationStyle = pagStyle
 }
+
+func (m *model) toggleSelectedSingle() {
+	m.prevDirection = ""
+	index := m.list.Index()
+	item, ok := m.list.SelectedItem().(item)
+	if !ok {
+		return
+	}
+	item.selected = !item.selected
+	m.list.SetItem(index, item)
+}
+
+func (m *model) toggleSelected(direction string) {
+	if m.prevDirection == "" {
+		m.prevDirection = direction
+	}
+
+	item, ok := m.list.SelectedItem().(item)
+	if !ok {
+		return
+	}
+
+	//index := m.list.Index()
+	index := m.list.Index()
+
+	if item.selected {
+		item.selected = false
+	} else if m.prevDirection == direction && !item.selected {
+		item.selected = true
+	} else {
+		m.prevDirection = ""
+	}
+
+	m.list.SetItem(index, item)
+
+	switch direction {
+	case "down":
+		m.list.CursorDown()
+	case "up":
+		m.list.CursorUp()
+	}
+}
+
+// used to retrieve selected items with their list view indexes
+type SelectedItem struct {
+	Index       int
+	Description string
+	Value       string
+}
+
+// index values, descriptions and full values of all selected items
+func (m *model) selectedItems() []SelectedItem {
+	selectedItems := []SelectedItem{}
+	for index, i := range m.list.Items() {
+		item, ok := i.(item)
+		if !ok {
+			continue
+		}
+		if item.selected {
+			selectedItems = append(
+				selectedItems,
+				SelectedItem{
+					Index:       index,
+					Description: item.descriptionBase,
+					Value:       item.titleFull,
+				},
+			)
+		}
+	}
+	return selectedItems
+}
+
+// iterate over the list items backwards so the indexes are not affected
+func (m *model) removeSelected() {
+	items := m.list.Items()
+	for i := len(items) - 1; i >= 0; i-- {
+		if item, ok := items[i].(item); ok && item.selected {
+			m.list.RemoveItem(i)
```

Comment:
```suggestion
			m.list.RemoveItem(i)
     break
```

Because one one is selected, no?

Reply:
This is probably just confusing langauge. Func is to remove all selected items that were selected as part of a multi-select, the removal of the current ''selected/chosen' item is handled outside of the func. I may reword.